src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatCopy = [...fromFormat];
+  const toFormatCopy = [...toFormat];
+
+  const separator = fromFormatCopy.pop();
+  const parts = date.split(separator);
+
+  const dateParts = {};
+
+  fromFormatCopy.forEach((part, index) => {
+    dateParts[part] = parts[index];
+  });
+
+  if (toFormatCopy.includes('YY') && dateParts['YYYY']) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  }
```

Comment:
The logic here assumes that 'YYYY' is always present in `fromFormat`, which might not be true. Ensure that the code can handle cases where 'YYYY' is absent.